plugins/httpapi/mso.py

```diff
@@ -187,6 +161,27 @@ def send_request(self, method, path, data=None):
             raise ConnectionError(json.dumps(self._verify_response(None, method, full_path, None)))
         return self._verify_response(response, method, full_path, rdata)
 
+    def set_connection_parameters(self):
+        connection_parameters = {}
+        for key in CONNECTION_KEYS:
+            if key == "login_domain":
+                value = self.params.get(key) if self.params.get(key) is not None else self.get_option(CONNECTION_MAP.get(key, key))
+                self.set_option(key, value)
+            else:
+                value = self.params.get(key) if self.params.get(key) is not None else self.connection.get_option(CONNECTION_MAP.get(key, key))
+                self.connection.set_option(CONNECTION_MAP.get(key, key), value)
+
+            connection_parameters[key] = value
+            if value != self.connection_parameters.get(key) and key in RESET_KEYS:
```

Comment:
Would it be good to have a condition to reset the connection if there's a switch from credentials used in the task to credentials used in the inventory? There may be a scenario where the credentials match in the inventory and the task.